sensor/kubernetes/eventpipeline/output/output_impl.go

```diff
@@ -61,27 +57,32 @@ func wrapSensorEvent(update *central.SensorEvent) *central.MsgFromSensor {
 // and sends the deployments (if needed) to Detector
 func (q *outputQueueImpl) runOutputQueue() {
 	for {
```

Comment:
Shouldn't the `innerQueue` and `forwardQueue` be defer-closed here?
```suggestion
func (q *outputQueueImpl) runOutputQueue() {
    defer close(q.innerQueue)
    defer close(q.forwardQueue)
	for {
```

Reply:
Closing the `q.forwardQueue` makes sense because `runOutputQueue` is the only goroutine writing to it.

Closing `q.innerQueue` doesn't make sense here. That would cause the exact panic this PR is fixing, just from another goroutine. A gorotuine could be processing a `Send` function about to write into the channel at the same time that this defer was called, resulting in a panic due to send on a close channel.

Reply:
Actually, closing `q.forwardQueue` also doesn't make sense (see panic in latest run https://storage.googleapis.com/origin-ci-test/pr-logs/pull/stackrox_stackrox/8120/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1713892608062787584/build-log.txt). The problem is that `Start()` and `Stop()` are going to be called multiple times if the connection restarts. So we shouldn't close the `q.forwardQueue` whenever `Stop()` was called, because `Start()` will be called later and a panic will happen.

Reply:
> Closing q.innerQueue doesn't make sense here. That would cause the exact panic this PR is fixing, just from another goroutine. A gorotuine could be processing a Send function about to write into the channel at the same time that this defer was called, resulting in a panic due to send on a close channel.


Yeah I recall seeing that problem before. I thin at some point would be nice to solve that problem but for now I agree it's better to not close the channels.

Reply:
Not sure I'd consider it a problem. There is no risk of leaking resources [as channels are garbage collected](https://groups.google.com/g/golang-nuts/c/pZwdYRGxCIk/m/rGr1D-uTQMEJ).